### PR TITLE
CASMINST-4963 CAN > CHN

### PR DESCRIPTION
**Pull request by lukebates123**
# Description

# Checklist Before Merging

- [ ] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [ ] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [ ] My commits or Pull-Request Title contain my JIRA information, or I do not have a JIRA.

[1]: https://github.com/Cray-HPE/docs-csm/blob/main/introduction/documentation_conventions.md#using-prompts
[2]: https://github.com/Cray-HPE/teams

### CODE REVIEW

File: upgrade/scripts/sls/csm_can_to_chn/sls_updates.py
Comment (trad511) [inline upgrade/scripts/sls/csm_can_to_chn/sls_updates.py:111]: Needs better messaging:  `CHN subnet not found in SLS.  To remove the CAN, first create a CHN.`

File: upgrade/scripts/sls/csm_can_to_chn/sls_updates.py
Comment (trad511) [inline upgrade/scripts/sls/csm_can_to_chn/sls_updates.py]: Better messaging.  `The current user network is set to {default_route}.  To remove the CAN this value must be set to CHN first.`

File: upgrade/scripts/sls/sls_can_to_chn.py
Comment (trad511) [inline upgrade/scripts/sls/sls_can_to_chn.py:151]: remove this if the function is being used.

**Review by trad511 — CHANGES_REQUESTED** [review]
Click prompt: https://click.palletsprojects.com/en/8.1.x/prompts/?highlight=prompt

Move ./scripts/operations/bifurcated_can/bican_route.py into the scripts/sls directory can/chn scripts and give it an edit/audit please.

**Comment by mharding-hpe** [comment]
This is still a draft, but I'm adding the "do not merge" label because I want to review this before we merge it (and have had bad luck with things being merged when I have only commented on this in my feedback but not added the label). This does not imply anything about my feelings as to the quality of the content here -- it just means that I don't know if I will have time to review this before the first time it will be considered for merging.

I will remove the label once it is marked ready for review, I've finished my reviews, and issues identified (if any) are addressed.

File: upgrade/scripts/bss/bss_remove_can.py
Comment (johren-hpe) [inline upgrade/scripts/bss/bss_remove_can.py:32]: It looks like these two numbered items carried over from sls_can_to_chn.py.   I don't think this script is doing those two things.

File: upgrade/scripts/bss/bss_remove_can.py
Comment (johren-hpe) [inline upgrade/scripts/bss/bss_remove_can.py:83]: Could there be cases where the ntp.allow section doesn't exist?   Or are you confident that will always be there?

Comment (johren-hpe) [reply upgrade/scripts/bss/bss_remove_can.py:83]: Is it possible that the CAN network (e.g. 10.103.11.0/25) but not the specific CAN IP (e.g. 10.103.11.10/25) exists in the ntp.allow section?

File: upgrade/scripts/bss/post-bootparameters.sh
Comment (johren-hpe) [inline upgrade/scripts/bss/post-bootparameters.sh:68]: remove debug?

File: upgrade/scripts/sls/csm_can_to_chn/sls_updates.py
Comment (johren-hpe) [inline upgrade/scripts/sls/csm_can_to_chn/sls_updates.py:71]: cut-and-paste error:  can_subnet should be cmn_subnet

File: upgrade/scripts/sls/csm_can_to_chn/sls_updates.py
Comment (johren-hpe) [inline upgrade/scripts/sls/csm_can_to_chn/sls_updates.py:82]: It scares me to have a default subnet here.  If they miss/ignore this warning and it moves on with the default, there will most likely be issues and it might take a while to figure out where the issue is.   It seems like we should instead fail if they don't provide that value on input.

**Review by trad511 — APPROVED** [review]

**Review by rustydb — APPROVED** [review]

**Review by gbaker-hpe — CHANGES_REQUESTED** [review]
We need to move the CHN upgrade script to be separate from the 1.3 upgrade directory.  The directory for these scripts will be operations/network/customer_accessible_networks/can_to_chn/scripts which is current staged in the move_network_docs_and_chn_procedure branch.

**Pull request closed without merging**
